media/server/gstplayer/source/tasks/generic/AttachSource.cpp

```diff
@@ -300,7 +300,19 @@ void AttachSource::switchAudioSource(GstCaps *caps, const std::string &strCaps)
         RIALTO_SERVER_LOG_WARN("SKIP switching audio source. Unknown mime type");
         return;
     }
-    RIALTO_SERVER_LOG_MIL("Switching audio source. New caps: %s", strCaps.c_str());
+    GstAppSrc *appSrc{GST_APP_SRC(m_context.streamInfo[m_attachedSource->getType()])};
+    std::string oldCapsStr;
+    GstCaps *oldCaps = m_gstWrapper->gstAppSrcGetCaps(appSrc);
+    if (oldCaps)
+    {
+        gchar *oldCapsCStr = m_gstWrapper->gstCapsToString(oldCaps);
+        oldCapsStr = std::string(oldCapsCStr);
+        m_glibWrapper->gFree(oldCapsCStr);
+        m_gstWrapper->gstCapsUnref(oldCaps);
+    }
+    RIALTO_SERVER_LOG_MIL("Switching audio source.");
```

Comment:
would be nice to see caps in this MIL log (at least when they are different).

Reply:
I'll change two logs below in next pull request